### PR TITLE
Fusion of Main and other MAM-js

### PR DESCRIPTION
**Pull request by EricBoucher22**

### CODE REVIEW

File: UDV-Core/examples/MAM/Demo.html [hunk 1]
```diff
@@ -5,20 +5,15 @@
...
+  <body>
+      <div id="viewerDiv"></div>
+      <img id="logoIMU" src="../data/img/logo-imu.png"></img>
+      <img id="logoLIRIS" src="../data/img/logo-liris.png"></img>
+      <script src="../../../../dist/udvcore.js"></script>
+      <script src="../../../src/Externals/dat.gui/dat.gui.min.js"></script>
+      <script src="../../../../dist/itowns/examples/js/GUI/GuiTools.js"></script>
+      <script src="../../../../dist/itowns/dist/itowns.js"></script>
+      <script src="../../../../dist/itowns/dist/debug.js"></script>
+      <script src="MAM.js"></script>
```
Comment (jailln) [inline UDV-Core/examples/MAM/Demo.html]: Please add a comment to explain how and when we use udvcore and itowns

File: UDV-Core/examples/MAM/MAM.js [hunk 2]
```diff
@@ -0,0 +1,316 @@
+//Working
```
Comment (jailln) [inline UDV-Core/examples/MAM/MAM.js]: Please add a comment stating that this code is inspired from itowns examples with a lin kto the corresponding example.

File: UDV-Core/examples/MAM/MAM.js [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+
+// Since the elevation layer use color textures, specify min/max z which is in
+// gray scale. Normally a multiplicative factor should allow to get the data at
+// the right scale but it is not done by the Open Data Grand Lyon
+const config = {
+    // materialOptions: {
+    //     useColorTextureElevation: true,
+    //     colorTextureElevationMinZ: 0,
+    //     colorTextureElevationMaxZ: 255,
+    // },
```
Comment (jailln) [inline UDV-Core/examples/MAM/MAM.js]: Please remove unused comments.

File: UDV-Core/examples/MAM/MAM.js [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+//         extent: extent,
+//         url: 'https://download.data.grandlyon.com/wms/grandlyon',
+//         networkOptions: { crossOrigin: 'anonymous' },
+//         protocol: 'wms',
+//         name: 'MNT2012_Altitude_10m_CC46',
+//         projection: 'EPSG:3946',
+//         heightMapWidth: 256,
+//         format: 'image/jpeg',
+//     },
+// });
```
Comment (jailln) [inline UDV-Core/examples/MAM/MAM.js]: Please remove unused comments.

File: UDV-Core/examples/MAM/MAM.js [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+//             if (!result) {
+//                 result = itowns.DEMUtils.getElevationValueAt(layer, contour[i], 0);
+//             }
+//             tile = [result.tile];
+//             altitudes.push(result.z + 2);
+//         }
+//         return altitudes;
+//     }
+//     return 0;
+// }
```
Comment (jailln) [inline UDV-Core/examples/MAM/MAM.js]: Please remove unused comments.

File: UDV-Core/examples/MAM/MAM.js [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+
+view.addLayer({
+    id:'WFS Bus',
+    type: 'geometry',
+    name: 'lyon_tcl_bus',
+    update: itowns.FeatureProcessing.update,
+    convert: itowns.Feature2Mesh.convert({
+        color: colorLine,
+      /*altitude: altitudeLine*/ }),
```
Comment (jailln) [inline UDV-Core/examples/MAM/MAM.js]: Please remove unused comments.

File: UDV-Core/examples/MAM/MAM.js [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+    return;
+  }
+
+/*  if (event.ctrlKey) {
+    // Même si event.key n'est pas 'Control' (par ex., 'a' is pressed),
+    // event.ctrlKey peut être true si la touche Ctrl est pressée dans le même temps.
+    alert(`Combinaison de ctrlKey + ${nomTouche}`);
+  } else {
+    alert(`Touche pressée ${nomTouche}`);
+  }*/
```
Comment (jailln) [inline UDV-Core/examples/MAM/MAM.js]: Please remove unused comments.

**Comment by EricBoucher22** [comment]
I remove the unwannted files, the data folder got pictures and logos so i let it in the project.

File: UDV-Core/examples/MAM/Demo.html (hunk 1 above)
Comment (EricBoucher22) [reply UDV-Core/examples/MAM/Demo.html]: udvcore is only use to create the extras windows (help and about), itowns is required to initialize the map (view, extent and position), create the controler and generate some layer by using some function and event.

Comment (jailln) [reply UDV-Core/examples/MAM/Demo.html]: I meant leave a comment in the code....

File: UDV-Core/examples/MAM/MAM.js [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+    enableRotation: false,
+    rotateSpeed : 1.5,
+    zoomInFactor : 0.04,
+    zoomOutFactor : 0.04,
+    maxPanSpeed : 5.0,
+    minPanSpeed : 0.01,
+};
+
+var useControlsForEditing = false;
+//var positionOnGlobe = { longitude: 4.820, latitude: 45.7402, altitude: 2895};
```
Comment (jailln) [inline UDV-Core/examples/MAM/MAM.js]: Please remove unused comments

File: UDV-Core/examples/MAM/MAM.js [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+        version: '1.3.0',
+        name: 'Ortho2009_vue_ensemble_16cm_CC46',
+        projection: 'EPSG:3946',
+        extent: extent,
+        format: 'image/jpeg',
+    },
+});
+
+
+//longitude: 4.820, latitude: 45.7402, altitude: 2895
```
Comment (jailln) [inline UDV-Core/examples/MAM/MAM.js]: ?

File: UDV-Core/examples/MAM/MAM.js [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+	}
+    return;
+  }
+
+  if (event.key === 'a') {
+	  //Advanced controller (etiding option)
+    if (confirm('Do you want to switch controller option ?\n Current Option '+((useControlsForEditing)? "Edit Setting" : "Regular Setting"))) {
+		useControlsForEditing = !useControlsForEditing; //Change Option
+    alert((useControlsForEditing)? "Edit Setting Activate" : "Regular Setting Activate");//Inform about new setting
+    } else {
```
Comment (jailln) [inline UDV-Core/examples/MAM/MAM.js]: If nothing is done in the else condition you can remove it.

Comment (EricBoucher22) [reply UDV-Core/examples/MAM/MAM.js]: Nothing happened in else condition (the else condition is removed).

File: UDV-Core/examples/MAM/MAM.js [hunk 11]
```diff
@@ -0,0 +1,287 @@
...
+aboutController.onFinishChange( function(value) { about.refresh(); } );
+
+// About subwindow
+helpController = datDotGUI.add( help, 'windowIsActive'
+                              ).name( "Help" ).listen();
+helpController.onFinishChange( function(value) { help.refresh(); });
+
+//datDotGUI.close();     // By default the dat.GUI controls are rolled up
+
+
```
Comment (jailln) [inline UDV-Core/examples/MAM/MAM.js]: Please use a consistent indentation in the following lines...

**Comment by jailln** [comment]
Please add a link to your demo with a short description (following the model of the other demos and examples) in the new welcome page of UDV :)

**Comment by jailln** [comment]
Thanks, but your demo is not an "Individual Modules and Extensions Demos", it is a "Demo", so could you put it in the right section of the page please ? :)

Regarding the description, this demo doesn't demonstrate any module of UDV and  I would add that you can also visualize and interact with layers representing Lyon bus lines and buildings (in addition to the maps you already mention).

**Comment by jailln** [comment]
Thanks; There is a bug with the switch in the menu of the WMS Image and WMS Pollution layers though. Could you correct that please before we can merge ?

**Comment by jailln** [comment]
There is one last comment from my first review that you need to resolve before I can merge:

"Please add a comment at the beginning of MAM.js saying that this code is inspired from itowns examples with a link to the corresponding example."

**Pull request merged**